### PR TITLE
Added support of Guzzle 7 in composer.json

### PR DESCRIPTION
**Pull request by TarasMatskovych**
|    Q          |   A
|-------------- | ------
| QA            | no

### Description
* Added support of `Guzzle ^7.0.1`

### CODE REVIEW

**Comment by Ocramius** [comment]
Seems like this library needs adjustments, in order to be compatible with `guzzlehttp/guzzle:^7`? Please check CI failures :)

**Review by Ocramius — CHANGES_REQUESTED** [review]
Seems like this library needs adjustments, in order to be compatible with `guzzlehttp/guzzle:^7`? Please check CI failures :)

File: src/Check/GuzzleHttpService.php [hunk 1]
```diff
@@ -173,7 +174,9 @@ private function createPsr7Request($url, $method, array $headers, $body)
             || $body instanceof Iterator
             || (is_object($body) && method_exists($body, '__toString'))
         ) {
-            return $request->withBody(stream_for($body));
+            return function_exists('GuzzleHttp\Psr7\stream_for') ?
```
Comment (Ocramius) [inline src/Check/GuzzleHttpService.php]: Is there no way to just use the new functions, and perhaps drop older guzzle versions?

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/Check/GuzzleHttpService.php (hunk 1 above)
Comment (froschdesign) [reply src/Check/GuzzleHttpService.php]: The current version of Guzzle contains:

```json
"php": "^7.2.5 || ^8.0",
```

https://github.com/guzzle/guzzle/blob/ee0a041b1760e6a53d2a39c8c34115adc2af2c79/composer.json#L54

Comment (TarasMatskovych) [reply src/Check/GuzzleHttpService.php]: So do i understand you correctly ? I have to remove from `composer.json` Guzzle ^5.3.3 and use only new functions in this places ?

Comment (Ocramius) [reply src/Check/GuzzleHttpService.php]: Correct - if `^6` works, we can keep `^6 || ^7`
Alternatively, just move to `^7`, which is also fine.

**Comment by Ocramius** [comment]
Lowest dependencies for `8.0` still failing - are we missing something? 


```

6) LaminasTest\Diagnostics\GuzzleHttpServiceTest::testGuzzleCheck with data set #5 (null, null, 404, 'Laminas\Diagnostics\Result\Fa...erface', 'PUT')
Error: Class "GuzzleHttp\Psr7\Message" not found
```

**Comment by Ocramius** [comment]
Ah, I see that bumping to `^7.2` worked, thanks @TarasMatskovych!

File: src/Check/GuzzleHttpService.php (hunk 1 above)
Comment (TarasMatskovych) [reply src/Check/GuzzleHttpService.php]: I updated minimum version of Guzzle to `7.2` for ability use new guzzle functions in the one way - now all checks are passed.

**Review by Ocramius — APPROVED** [review]
:ship: thanks @TarasMatskovych!

**Comment by Ocramius** [comment]
Tagged as `1.14.0`

**Pull request merged**